File: test/scripts/13-memory-pooling/incorrect.js
const N = 100_000;
const BUF = 4096;
const DURATION_SEC = parseInt(process.env.DURATION_SEC || '300', 10);
let laps = 0;
const start = Date.now();
console.log('INCORRECT: new Buffer each call');
function run() {
  if (Date.now() - start >= DURATION_SEC * 1000) {
    console.log('Stopped:', laps, 'laps');
    return;
  }
  for (let i = 0; i < N; i++) {
    const buf = Buffer.alloc(BUF);
    buf[0] = 1;
  }
  laps++;
  setImmediate(run);
}
run();
process.on('SIGINT', () => process.exit(0));

File: test/scripts/13-memory-pooling/incorrect.py
#!/usr/bin/env python3
"""
INCORRECT: Allocate buffer each time - no pooling.
Resume: "Use Memory Pooling"
"""
import os
import signal
import time
N = 100_000
BUF_SIZE = 4096
DURATION_SEC = int(os.environ.get("DURATION_SEC", "300"))
stop = False

def main():
    global stop
    print("INCORRECT: new byte[] each call")
    signal.signal(signal.SIGINT, lambda s, f: globals().__setitem__('stop', True))
    signal.signal(signal.SIGTERM, lambda s, f: globals().__setitem__('stop', True))
    start = time.perf_counter()
    laps = 0
    while not stop and (time.perf_counter() - start) < DURATION_SEC:
        for _ in range(N):
            buf = bytearray(BUF_SIZE)
            buf[0] = 1
        laps += 1
    print(f"Stopped: {laps} laps")

if __name__ == "__main__":
    main()
